### ISSUE TITLE
AutoEncoder using Deconvolution Error

### ISSUE THREAD

**Issue by OspreyAction** [Microsoft/CNTK#881]
Hello,

I tried a couple of weeks to use CNTK for an autoencoder using deconvolution, training a model to minimize MSE.

However, training does not work and all gray values of output image are 0 like #701 .

Here are my cntk, ndl and output text files.

Thank you for any suggestion.

[04_DeConv_ndl.txt](https://github.com/Microsoft/CNTK/files/502363/04_DeConv_ndl.txt)
[04_DeConv_cntk.txt](https://github.com/Microsoft/CNTK/files/502364/04_DeConv_cntk.txt)
[out.txt](https://github.com/Microsoft/CNTK/files/502365/out.txt)

**Comment by e-thereal** [comment]
Have you tried a simpler model with fewer layers first? Also, you are training for only one epoch, which is likely not enough. Try setting maxEpochs to some higher value, say, 100 and check that the MSE is going down. If not, you might have to change the initial values or the learning rate, by I bet that you just need to train for more epochs, or start with a simpler model first.

**Comment by OspreyAction** [comment]
Thank you very much for your comment.

I haven't tried a simpler model  because I am busy at other tasks. I will try this if I can get relaxed.

Yes, I trained for only one epoch. This is because I wanted to examine whether deconvolution network in CNTK actually works. I thought that if deconvolution network in CNTK works, all gray values of output image won't be 0, although my network is not viable and training doesn't go to minimum error at somewhere. I just wanted to see gray values not 0.

**Comment by e-thereal** [comment]
You are right. Assuming that the output layer is linear, you would expect the outputs to be at least slightly different from zero. Sorry I can't help you further.

**Comment by pkranen** [comment]
We have a new example for deconvolution which builds a simple auto encoder for MNIST images in both Python and BrainScript. Please check these links:

[Examples/Image/GettingStarted/07_Deconvolution_PY.py](https://github.com/Microsoft/CNTK/blob/master/Examples/Image/GettingStarted/07_Deconvolution_PY.py)

[https://github.com/Microsoft/CNTK/wiki/Image-Auto-Encoder-Using-Deconvolution-And-Unpooling](https://github.com/Microsoft/CNTK/wiki/Image-Auto-Encoder-Using-Deconvolution-And-Unpooling)

I close this issue, feel free to reopen.
